### PR TITLE
Update tikv key metric

### PR DESCRIPTION
**Pull request by hunterlxt**
Signed-off-by: TXXT <hunterlxt@live.com>

### What is changed?

Update tikv key metric

### Any related PRs or issues?

close https://github.com/pingcap/docs-cn/issues/2801

### Which version does your change affect?

4.0 master

### CODE REVIEW

**Comment by dcalvin** [comment]
@ran-huang PTAL.

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 1]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Rocksdb | tikv_engine_wal_file_sync_micro_seconds | Histogram of WAL file sync micros. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_sst_read_micros | Histogram of SST read micros. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_compaction_time | Histogram of compaction time. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_block_cache_size_bytes | Usage of each column families' block cache. The metric has 2 labels `db` and `cf`. |
+Rocksdb | tikv_engine_compaction_reason | Number of compaction reason. The metric has 3 labels `db`, `cf` and `reason`.  |
+Rocksdb | tikv_engine_cache_efficiency | Efficiency of rocksdb's block cache. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_memtable_efficiency | Hit and miss of memtable. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_bloom_efficiency | Efficiency of rocksdb's bloom filter. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_estimate_num_keys | Estimate num keys of each column families. The metric has 2 labels `db` and `cf`. |
+Rocksdb | tikv_engine_compaction_flow_bytes | Bytes of read/written during compaction |
```
Comment (ran-huang) [inline content/docs/dev/tasks/monitor/key-metrics.md:107]:
```suggestion
Rocksdb | tikv_engine_compaction_flow_bytes | Bytes of read/write during compaction |
```

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 2]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Rocksdb | tikv_engine_event_total | Number of engine events. The metric has 3 labels `db`, `cf` and `type`. |
+Rocksdb | tikv_engine_wal_file_sync_micro_seconds | Histogram of WAL file sync micros. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_sst_read_micros | Histogram of SST read micros. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_compaction_time | Histogram of compaction time. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_block_cache_size_bytes | Usage of each column families' block cache. The metric has 2 labels `db` and `cf`. |
+Rocksdb | tikv_engine_compaction_reason | Number of compaction reason. The metric has 3 labels `db`, `cf` and `reason`.  |
+Rocksdb | tikv_engine_cache_efficiency | Efficiency of rocksdb's block cache. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_memtable_efficiency | Hit and miss of memtable. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_bloom_efficiency | Efficiency of rocksdb's bloom filter. The metric has 2 labels `db` and `type`. |
```
Comment (ran-huang) [inline content/docs/dev/tasks/monitor/key-metrics.md:105]:
```suggestion
Rocksdb | tikv_engine_bloom_efficiency | Efficiency of RocksDB's bloom filter. The metric has 2 labels `db` and `type`. |
```

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 3]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Rocksdb | tikv_engine_compaction_flow_bytes | Bytes of read/written during compaction |
+Rocksdb | tikv_engine_bytes_per_read | Histogram of bytes per read. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_read_amp_flow_bytes | Bytes of read amplification. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_bytes_per_write | tikv_engine_bytes_per_write. The metric has 2 labels `db` and `type`. |
+Rocksdb | tikv_engine_num_snapshots | Number of unreleased snapshots. The metric has label `db`. |
+Rocksdb | tikv_engine_pending_compaction_bytes | Pending compaction bytes. The metric has 2 labels `db` and `cf`. |
+Rocksdb | tikv_engine_num_files_at_level | Number of files at each level. The metric has 3 labels `db`, `cf` and `level`. |
+Rocksdb | tikv_engine_compression_ratio | Compression ratio at different levels. The metric has 3 labels `db`, `cf` and `level`. |
+Rocksdb | tikv_engine_oldest_snapshot_duration | Oldest unreleased snapshot duration in seconds. The metric has label `db`. |
+Rocksdb | tikv_engine_write_stall_reason | QPS of each reason which cause tikv write stall. The metric has 2 labels `db` and `type`. |
```
Comment (ran-huang) [inline content/docs/dev/tasks/monitor/key-metrics.md:116]:
```suggestion
Rocksdb | tikv_engine_write_stall_reason | QPS of each reason which causes TiKV write stall. The metric has 2 labels `db` and `type`. |
```

**Comment by ran-huang** [comment]
Friendly ping @hunterlxt 😃

**Comment by hunterlxt** [comment]
@ran-huang Thanks for review, I'll push commit later.

**Comment by ran-huang** [comment]
Another friendly ping @hunterlxt

**Comment by dcalvin** [comment]
@hunterlxt Please fix the DCO.

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 4]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Scheduler | tikv_scheduler_contex_total | Total number of pending commands. The scheduler receives commands from clients, executes them against the MVCC layer storage engine. |
+Scheduler | tikv_scheduler_stage_total | Total number of commands on each stage. The metric has 2 labels `type` and `stage`. `stage` represents the stage of executed command like "read_finish", "async_snapshot_err", "snapshot", etc. |
+Scheduler | tikv_scheduler_commands_pri_total | Total count of different priority commands. The metric has label `priority`. |
+Server | tikv_server_grpc_resp_batch_size | grpc batch size of gRPC responses. |
+Server | tikv_server_report_failure_msg_total | Total number of reporting failure messages. The metric has 2 labels, `type` represents the failure type and `store_id` represents the destination peer store id. |
+Server | tikv_server_raft_message_flush_total | Total number of raft messages flushed immediately. |
+Server | tikv_server_raft_message_recv_total | Total number of raft messages received. |
+Server | tikv_region_written_keys | Histogram of keys written for regions. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:34]: Fixed

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 5]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Raft | tikv_raftstore_commit_log_duration_seconds | Bucketed histogram of peer commits logs duration. |
+Raft | tikv_raftstore_raft_ready_handled_total | Total number of raft ready handled. The metric has label `type`. |
+Raft | tikv_raftstore_raft_process_duration_secs | Bucketed histogram of peer processing raft duration. The metric has label `type`. |
+Raft | tikv_raftstore_event_duration | Duration of raft store events. The metric has label `type`. |
+Raft | tikv_raftstore_raft_sent_message_total | Total number of raft ready sent messages. The metric has label `type`. |
+Raft | tikv_raftstore_raft_dropped_message_total | Total number of raft dropped messages. The metric has label `type`. |
+Raft | tikv_raftstore_apply_proposal | The count of proposals sent by a region at once. |
+Raft | tikv_raftstore_proposal_total | Total number of proposal made. The metric has label `type`. |
+Raft | tikv_raftstore_request_wait_time_duration_secs | Bucketed histogram of request wait time duration. |
+Raft | tikv_raftstore_propose_log_size | Bucketed histogram of peer proposing log size. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:50]: the record format is Bucketed histogram, and its content is the size of each proposing log. Do you mean to explain what is called propose?

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 6]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Raft | tikv_raftstore_apply_wait_time_duration_secs | Bucketed histogram of apply task wait time duration. |
+Raft | tikv_raftstore_admin_cmd_total | Total number of admin cmd processed. The metric has 2 labels `type` and `status`. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:52]: Fixed

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 7]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Raft | tikv_raftstore_check_split_total | Total number of raftstore split check. The metric has label `type`. |
+Raft | tikv_raftstore_check_split_duration_seconds | Bucketed histogram of raftstore split check duration. |
+Raft | tikv_raftstore_local_read_reject_total | Total number of rejections from the local reader. The metric has label `reason` which represents rejection reason. |
+Raft | tikv_raftstore_snapshot_duration_seconds | Bucketed histogram of raftstore snapshot process duration. The metric has label `type`. |
+Raft | tikv_raftstore_snapshot_traffic_total | Total number of raftstore snapshot traffic. The metric has label `type`. |
+Raft | tikv_raftstore_local_read_executed_requests | Total number of requests directly executed by local reader. |
+Coprocessor | tikv_coprocessor_request_duration_seconds | Bucketed histogram of coprocessor request duration. The metric has label `req`. |
+Coprocessor | tikv_coprocessor_request_error | Total number of push down request error. The metric has label `reason`. |
+Coprocessor | tikv_coprocessor_scan_keys | Bucketed histogram of coprocessor per request scan keys. The metric has label `req` which represents the tag of requests. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:61]: Fixed

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 8]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Coprocessor | tikv_coprocessor_rocksdb_perf | Total number of RocksDB internal operations from PerfContext. The metric has 2 labels `req` and `metric`. `req` represents the tag of requests and `metric` is perf metric like "block_cache_hit_count", "block_read_count", "encrypt_data_nanos", etc. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:62]: Fixed

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 9]
```diff
@@ -7,52 +7,111 @@ menu:
...
+Storage | tikv_storage_mvcc_gc_delete_versions | Histogram of versions deleted by gc for each key. |
+Storage | tikv_storage_mvcc_conflict_counter | Total number of conflict error. The metric has label `type`. |
+Storage | tikv_storage_mvcc_duplicate_cmd_counter | Total number of duplicated commands. The metric has label `type`. |
+Storage | tikv_storage_mvcc_check_txn_status | Counter of different results of check_txn_status. The metric has label `type`. |
+Storage | tikv_storage_command_total | Total number of commands received. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_duration_seconds | Bucketed histogram of processing successful asynchronous requests. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_total | Total number of engine asynchronous requests. The metric has 2 labels `type` and `status`. |
+GC | tikv_gcworker_gc_task_fail_vec | Counter of gc tasks that is failed. The metric has label `task`. |
+GC | tikv_gcworker_gc_task_duration_vec | Duration of gc tasks execution. The metric has label `task`. |
+GC | tikv_gcworker_gc_task_duration_vec | Duration of gc tasks execution. The metric has label `task`. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:75]: Fixed

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 10]
```diff
@@ -7,52 +7,111 @@ menu:
...
+GC | tikv_gcworker_autogc_processed_regions | Processed regions by auto gc. The metric has label `type`. |
+GC | tikv_gcworker_autogc_safe_point | Safe point used for auto gc. The metric has label `type`. |
+Snapshot | tikv_snapshot_size | Size of snapshot. |
+Snapshot | tikv_snapshot_kv_count | Total number of kv in snapshot |
+Snapshot | tikv_worker_handled_task_total | Total number of worker handled tasks. The metric has label `name`. |
+Snapshot | tikv_worker_pending_task_total | Current worker pending + running tasks. The metric has label `name`.|
+Snapshot | tikv_futurepool_handled_task_total | Total number of future_pool handled tasks. The metric has label `name`. |
+Snapshot | tikv_snapshot_ingest_sst_duration_seconds | Bucketed histogram of rocksdb ingestion durations |
+Snapshot | tikv_futurepool_pending_task_total | Current future_pool pending + running tasks. The metric has label `name`. |
+Rocksdb | tikv_engine_get_served | Get queries served by engine. The metric has 2 labels `db` and `type`. |
```
[the remark this answers is not in the document]
Comment (hunterlxt) [reply content/docs/dev/tasks/monitor/key-metrics.md:86]: just queries served in rocksdb, i think "queries served by engine" is enough? @ran-huang

File: content/docs/dev/tasks/monitor/key-metrics.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ran-huang) [reply content/docs/dev/tasks/monitor/key-metrics.md:50]: How about `Bucketed histogram of the size of each peer proposing log.` This expression is clearer.

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 11]
```diff
@@ -7,52 +7,110 @@ menu:
...
+Storage | tikv_storage_mvcc_gc_delete_versions | Histogram of versions deleted by GC for each key. |
+Storage | tikv_storage_mvcc_conflict_counter | Total number of conflict error. The metric has label `type`. |
+Storage | tikv_storage_mvcc_duplicate_cmd_counter | Total number of duplicated commands. The metric has label `type`. |
+Storage | tikv_storage_mvcc_check_txn_status | Counter of different results of `check_txn_status`. The metric has label `type`. |
+Storage | tikv_storage_command_total | Total number of commands received. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_duration_seconds | Bucketed histogram of processing successful asynchronous requests. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_total | Total number of engine asynchronous requests. The metric has 2 labels `type` and `status`. |
+GC | tikv_gcworker_gc_task_fail_vec | Counter of failed GC tasks. The metric has label `task`. |
+GC | tikv_gcworker_gc_task_duration_vec | Duration of gc tasks execution. The metric has label `task`. |
+GC | tikv_gcworker_gc_keys | Counter of keys affected during gc. The metric has two labels `cf` and `tag`. |
```
Comment (ran-huang) [inline content/docs/dev/tasks/monitor/key-metrics.md:75]:
```suggestion
GC | tikv_gcworker_gc_keys | Counter of keys affected during GC. The metric has two labels `cf` and `tag`. |
```

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 12]
```diff
@@ -7,52 +7,110 @@ menu:
...
+Storage | tikv_storage_mvcc_versions | Histogram of versions for each key. |
+Storage | tikv_storage_mvcc_gc_delete_versions | Histogram of versions deleted by GC for each key. |
+Storage | tikv_storage_mvcc_conflict_counter | Total number of conflict error. The metric has label `type`. |
+Storage | tikv_storage_mvcc_duplicate_cmd_counter | Total number of duplicated commands. The metric has label `type`. |
+Storage | tikv_storage_mvcc_check_txn_status | Counter of different results of `check_txn_status`. The metric has label `type`. |
+Storage | tikv_storage_command_total | Total number of commands received. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_duration_seconds | Bucketed histogram of processing successful asynchronous requests. The metric has label `type`. |
+Storage | tikv_storage_engine_async_request_total | Total number of engine asynchronous requests. The metric has 2 labels `type` and `status`. |
+GC | tikv_gcworker_gc_task_fail_vec | Counter of failed GC tasks. The metric has label `task`. |
+GC | tikv_gcworker_gc_task_duration_vec | Duration of gc tasks execution. The metric has label `task`. |
```
Comment (ran-huang) [inline content/docs/dev/tasks/monitor/key-metrics.md:74]:
```suggestion
GC | tikv_gcworker_gc_task_duration_vec | Duration of GC tasks execution. The metric has a label `task`. |
```

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 13]
```diff
@@ -7,52 +7,110 @@ menu:
...
+Raft | tikv_raftstore_event_duration | Duration of raft store events. The metric has label `type`. |
+Raft | tikv_raftstore_raft_sent_message_total | Total number of messages sent by Raft ready. The metric has label `type`. |
+Raft | tikv_raftstore_raft_dropped_message_total | Total number of messages dropped by Raft. The metric has label `type`. |
+Raft | tikv_raftstore_apply_proposal | The count of proposals sent by a region at once. |
+Raft | tikv_raftstore_proposal_total | Total number of proposals made. The metric has label `type`. |
+Raft | tikv_raftstore_request_wait_time_duration_secs | Bucketed histogram of request wait time duration. |
+Raft | tikv_raftstore_propose_log_size | Bucketed histogram of peer proposing log size. |
+Raft | tikv_raftstore_apply_wait_time_duration_secs | Bucketed histogram of apply task wait time duration. |
+Raft | tikv_raftstore_admin_cmd_total | Total number of admin command processed. The metric has 2 labels `type` and `status`. |
+Raft | tikv_raftstore_check_split_total | Total number of raftstore split check. The metric has label `type`. |
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply content/docs/dev/tasks/monitor/key-metrics.md:53]: Suggested use: has a `type` label

File: content/docs/dev/tasks/monitor/key-metrics.md [hunk 14]
```diff
@@ -7,52 +7,110 @@ menu:
...
-System Info | Memory Available | the size of the available memory |
-System Info | Network Traffic | the statistics of the network traffic |
-System Info | TCP Retrans | the statistics about network monitoring and TCP |
-System Info | IO Util | the disk usage ratio, 100% at a maximum; generally you need to consider adding a new node when the usage ratio is up to 80% ~ 90% |
\ No newline at end of file
+Cluster | tikv_store_size_bytes | The size of storage. The metric has a `type` label (eg: "capacity", "available"). |
+gRPC | tikv_grpc_msg_duration_seconds | Bucketed histogram of gRPC server messages. The metric has a `type` label which represents the type of the server message. You can count the metric and calculate the QPS. |
+gRPC | tikv_grpc_msg_fail_total | The total number of gRPC message handling failure. The metric has a `type` label which represents gRPC message type. |
+GRPC | grpc batch size of gRPC requests | grpc batch size of gRPC requests. |
+Scheduler | tikv_scheduler_too_busy_total | The total count of too busy schedulers. The metric has a `type` label which represents the scheduler type. |
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply content/docs/dev/tasks/monitor/key-metrics.md:26]: both are correct.

**Comment by dcalvin** [comment]
> LGTM.
> @dcalvin Shall we involve a tech review?

I think it's good in this case

**Pull request merged**